### DISCUSSION TITLE
Getting typeError on get_api_key()

### DISCUSSION THREAD

**Discussion by meetFarmanUllah** [vanna-ai/vanna#381] [Q&A] [answered]
I am getting a type error when getting the API key, I am receiving a code on email after entering my code got an error. I am following the same steps shown in the documentation.

Here is my code
api_key = vn.get_api_key('my-example@yahoo.com')
vn.set_api_key(api_key)

below is the error

Check your email for the code and enter it here: SSVZNL # successfully entering code from email
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[14], line 1
----> 1 api_key = vn.get_api_key('my-example@yahoo.com')
      2 vn.set_api_key(api_key)

File C:\ProgramData\anaconda3\lib\site-packages\vanna\__init__.py:123, in get_api_key(email, otp_code)
    120 if "result" not in d:
    121     raise OTPCodeError("Error verifying OTP code.")
--> 123 key = ApiKey(**d["result"])
    125 if key is None:
    126     raise OTPCodeError("Error verifying OTP code.")

TypeError: vanna.types.ApiKey() argument after ** must be a mapping, not NoneType

any solution for this or i am doing anything wrong?

**Comment by zainhoda** [comment, marked as answer]
Hi there! It looks like you're following some deprecated documentation.

Please use the docs from here:
https://vanna.ai/docs/postgres-openai-vanna-vannadb/

You can change your database and other preferences by clicking on the various options.

**Reply by meetFarmanUllah** [reply]
Can you share the updated documentation for creating and training a model? i don't know why but i found the documentation a bit confusing.
Thank you for your help
